tests/torch/accuracy_aware_training/test_training_loop.py

```diff
@@ -119,3 +129,203 @@ def configure_optimizers_fn():
+        return optimizer, None
+
+    early_stopping_training_loop = EarlyExitCompressionTrainingLoop(config, compression_ctrl)
+    model = early_stopping_training_loop.run(model,
+                                             train_epoch_fn=train_fn,
+                                             validate_fn=partial(validate_fn, train_loader=train_loader),
+                                             configure_optimizers_fn=configure_optimizers_fn)
+    original_model_accuracy = model.original_model_accuracy
+    compressed_model_accuracy = validate_fn(model, train_loader=train_loader)
+
+    assert (original_model_accuracy - compressed_model_accuracy) * 100 <= max_accuracy_degradation
+
+
+@pytest.mark.parametrize(
+    ('max_accuracy_degradation', 'exit_epoch_number'),
+    ((1.0, 6), (30.0, 10))
+)
+def test_early_exit_with_mock_validation(max_accuracy_degradation, exit_epoch_number,
+                                         maximal_total_epochs=100):
+    epoch_counter = 0
+
+    def mock_validate_fn(model, init_step=False, epoch=0):
+        original_metric = 0.85
+        if init_step:
+            return original_metric
+        nonlocal epoch_counter
+        epoch_counter = epoch
+        return original_metric * (1 - 0.01 * max_accuracy_degradation) * (epoch / exit_epoch_number)
+
+    config = get_quantization_config_without_range_init(LeNet.INPUT_SIZE[-1])
+
+    params = {
+        "maximal_accuracy_degradation": max_accuracy_degradation,
+        "maximal_total_epochs": maximal_total_epochs
+    }
+    accuracy_aware_config = {
+        "accuracy_aware_training": {
+            "mode": "early_exit",
+            "params": params
+        }
+    }
+
+    config.update(accuracy_aware_config)
+
+    train_loader = create_ones_mock_dataloader(config, num_samples=10)
+    model = LeNet()
+
+    config = register_default_init_args(config,
+                                        train_loader=train_loader,
+                                        model_eval_fn=partial(mock_validate_fn, init_step=True))
+
+    model, compression_ctrl = create_compressed_model_and_algo_for_test(model, config)
+
+    def train_fn(compression_ctrl, model, epoch, optimizer, lr_scheduler,
+                 train_loader=train_loader):
+        pass
+
+    def configure_optimizers_fn():
+        return None, None
+
+    early_stopping_training_loop = EarlyExitCompressionTrainingLoop(config, compression_ctrl,
+                                                                    dump_checkpoints=False)
+    model = early_stopping_training_loop.run(model,
+                                             train_epoch_fn=train_fn,
+                                             validate_fn=partial(mock_validate_fn),
+                                             configure_optimizers_fn=configure_optimizers_fn)
+    # Epoch number starts from 0
+    assert epoch_counter == exit_epoch_number
+
+
+@pytest.mark.parametrize(('aa_config', 'must_raise'), (
```

Comment:
You could move the config validation tests to another file. This one tests training loop functionality given its naming.

Reply:
Moved to test_accuracy_aware_config